docs/pages/project/contributing/contributing-cli.md

```diff
@@ -0,0 +1,134 @@
+
+**Logging**
+
+For logs, `mesheryctl` uses [Logrus](https://github.com/sirupsen/logrus). Going through the docs and understanding the different [log-levels](https://github.com/sirupsen/logrus#level-logging) will help a lot.
+
+**Linting**
+
+`mesheryctl` uses [golangci-lint](https://github.com/golangci/golangci-lint). See the .github/workflow/ci.yaml for syntax used during Meshery's build process.
+
+**Unit Tests**
+
+Unit test code coverage reports can be found - _tbd_.
+
+# Meshery CLI Style Guide
+
+These guidelines are a collection of principles and conventions that need to be followed while designing mesheryctl commands. `mesheryctl` might be the interface that the users first have with Meshery. As such, `mesheryctl` needs to provide a great UX.
+
+The following principles should be taken in mind while designing `mesheryctl` commands-
+
+## Design Principles
+
+**1. Consistency is quality.**
+
+- _Consistency of interaction drives a *quality* user experience. Whether that experience is delightful or painful is a related, but separate consideration. Meshery's behavior of user interactions should be consistent even when their user experience is poor._
+
+**2. Intuitive user experiences feel natural to users.**
+
+- _When being designed, each of Meshery's user experiences should be examined first from the user's perspective. Design user experiences that are familiar._
+
+**3. Design for brevity.**
+
+- _Avoid long commands with chained series of flags._
+
+**4. Design with automated testing in mind.**
+
+- _Provide possibility to specify output format as json (-o json) for easy inspection of command response._
+
+Part of delivering a great user experience is providing intuitive interfaces. In the case of `mesheryctl` takes inspiration from and delivers similar user experiences as popular CLIs do in this ecosystem, like `kubectl` and `docker`. Here is relevant `kubectl` information to reference - [Kubectl SIG CLI Community Meeting Minutes](https://docs.google.com/document/u/2/d/1r0YElcXt6G5mOWxwZiXgGu_X6he3F--wKwg-9UBc29I/edit#), [contributing to kubectl](https://github.com/kubernetes/community/blob/master/sig-cli/CONTRIBUTING.md), [code](https://github.com/kubernetes/kubernetes/tree/master/pkg/kubectl/cmd/config).
+
+Command structure and command behavior should be designed in such a way that they are intuitive. Users should ideally be able to understand what a command is used for without having to extensively go through the documentation. For example, `mesheryctl pattern apply -f <pattern name>` requires no further clarification as it is evident that the command will apply the pattern specified.
+
+Consistency is key when designing intuitive interfaces. Although `mesheryctl perf run -f <performance profile name>` may sound more intuitive, users who are experienced in using the CLI will prefer the consistant verb `apply` over `run`. This will also ensure a consistent command language making memorizing easier.
+
+** Flags **
```

Comment:
Was this supposed to be bold?

Reply:
Yes, this "Flags" is intentional. We need more sections with style "Commands" and "Subcommands".

Reply:
But, you're right that in the doc's current state, it's a little out of place.